src/RescriptCore.res

```diff
@@ -64,6 +64,11 @@ type null<+'a> = Js.null<'a>
 
 type undefined<+'a> = Js.undefined<'a>
 
-type nullable<+'a> = Js.nullable<'a>
+// We're intentionally only bringing the constructors of `nullable` into scope
+// here. Reason is if we do the same for `null`, we'll get issues with inference
+// in the global scope, and we prioritize nullable higher because that covers
+// more cases.
+@unboxed
+type nullable<+'a> = Js.nullable<'a> = Value('a) | @as(null) Null | @as(undefined) Undefined
```

Comment:
Good choice! It would have been even better for usability if this was structurally typed, I think, but I assume there are good reasons for using plain variants instead. The discussion would also be out-of-scope for this repo anyway. Given that it is nominally typed, giving priority to `nullable` seems like the optimal decision.